terraform/modules/kubernetes/monitoring.tf

```diff
@@ -0,0 +1,251 @@
+  count = var.enable_alerting ? (var.deploy_azure_backing_services ? 1 : 0 ) : 0
+
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-metricalert-cpu"
+  resource_group_name = data.azurerm_resource_group.backing-service-resource-group[0].name
+  scopes              = [azurerm_postgresql_flexible_server.postgres-server[0].id]
+  description         = "Action will be triggered when cpu use is greater than 75%"
+
+  criteria {
+    metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+    metric_name      = "cpu_percent"
+    aggregation      = "Average"
+    operator         = "GreaterThan"
+    threshold        = 75
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.main[0].id
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "postgres_storage" {
+  count = var.enable_alerting ? (var.deploy_azure_backing_services ? 1 : 0 ) : 0
+
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-metricalert-storage"
+  resource_group_name = data.azurerm_resource_group.backing-service-resource-group[0].name
+  scopes              = [azurerm_postgresql_flexible_server.postgres-server[0].id]
+  description         = "Action will be triggered when storage use is greater than 75%"
+
+  criteria {
+    metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+    metric_name      = "storage_percent"
+    aggregation      = "Average"
+    operator         = "GreaterThan"
+    threshold        = 75
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.main[0].id
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "redis_memory" {
+  count = var.enable_alerting ? (var.deploy_azure_backing_services ? 1 : 0 ) : 0
+
+  name                = "${azurerm_redis_cache.redis-queue[0].name}-metricalert-memory"
+  resource_group_name = data.azurerm_resource_group.backing-service-resource-group[0].name
+  scopes              = [azurerm_redis_cache.redis-queue[0].id]
+  description         = "Action will be triggered when memory use is greater than 75%"
+
+  criteria {
+    metric_namespace = "Microsoft.Cache/redis"
+    metric_name      = "allusedmemorypercentage"
+    aggregation      = "Average"
+    operator         = "GreaterThan"
+    threshold        = 75
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.main[0].id
+  }
+
+  lifecycle {
+    ignore_changes = [
+      tags
+    ]
+  }
+}
+
+# FD/CDN is in the prod subscription for all envs
+# and a built in a different terraform run
+
+locals {
+  cdn_profile = "s189p01-apply-svc-domains-fd"
+  cdn_rg      = "s189p01-applydomains-rg"
+# set to below for testing in dev review
+#  cdn_profile = "s189d01-fdsdomains-fd"
+#  cdn_rg      = "s189d01-fddomains-rg"
+}
+
+
+data "azurerm_cdn_frontdoor_profile" "svc_domain" {
```

Comment:
Since the front door is created in https://github.com/DFE-Digital/apply-for-teacher-training/tree/main/terraform/custom_domains/infrastructure, should we add the monitoring there for clarity?

Reply:
although I guess you could enable cdn monitoring for all envs if creating those alerts from custom_domains

Reply:
moved fd monitoring to fd terraform